src/main/java/org/spongepowered/api/text/Text.java

```diff
@@ -319,23 +318,29 @@ public String toString() {
 
     /**
      * Represents a {@link Text} placeholder that can be replaced with another
-     * Text by {@link Texts#format(Text, Map)}.
+     * Text by {@link Texts#format(Text, Function)}.
      *
      * @see TextBuilder.Placeholder
      */
     public static class Placeholder extends Text {
```

Comment:
There is a simple but important fact, that you have to (de-)serialize it.
I took a lot of time evaluating the fact that I could externalize the (de-)serialization. But then I would end up duplicating the entire Text API. Starting with TextRepresentation.fromJSON -> fromJSONWithPlaceholder/fromJSONAsTemplate via Texts.of() -> Texts.placehodlerOf() to the format call if I don't want to replace everything in one run.

For me the most important part is the serialization. The feature must be available! There shouldn't be any structural differences between the template text (string/json) and a real text. These would only complicate the plugin configuration and will be an annoyance for server owners.

Of course this is a deviation from vanilla text classes, but IMO it is justified, because it is small addition, has no impact on client-server communication or any other side effects and grants a lot of flexibility for plugin authors.